src/shared/i18next.ts

```diff
@@ -60,6 +60,36 @@ const resources = {
   pt,
 };
 
+const languageNames = {
+  en: "English",
+  ru: "Русский",
+  zh: "中文",
+  es: "Español",
+  eu: "Euskara",
+  bg: "Български",
+  nl: "Nederlands",
+  fi: "Suomi",
+  fr: "Français",
+  el: "Ελληνικά",
+  ko: "한국어",
+  pl: "Polski",
+  ar: "العربية",
+  eo: "Esperanto",
+  de: "Deutsch",
+  gl: "Galego",
+  it: "Italiano",
+  ja: "日本語",
+  km: "ភាសាខ្មែរ",
+  nb_NO: "Norsk (Bokmål)",
+  zh_Hant: "文言",
+  fa: "فارسی",
+  id: "Bahasa Indonesia",
+  mnc: "ᠮᠠᠨᠵᡠ ᡤᡳᠰᡠᠨ",
+  sv: "Svenska",
+  vi: "Tiếng Việt",
+  pt: "Português",
+};
+
```

Comment:
I tried some libraries for this, but none of them worked well, partly because our language identifiers are in different formats. And `mnc` (Manchu) is essentially a dead language with [20 native speakers](https://en.wikipedia.org/wiki/Manchu_language), so we can easily remove it.